### ISSUE TITLE
Truncation of labels on datatables leads to labels changed to '...'

### ISSUE THREAD

**Issue by canajun2eh** [piwik/piwik#6821]
Issue #1:
As can be seen in the attached screen shot (under issue #2, below), the Visitor Location (City) widget is not always displaying the complete location.  To the right of the country flag, one can sometimes see three dots instead of the actual location.  The display will randomly begin to show the correct location, only to revert to showing the three dots at some other random time.

When you hover over the country flag, the tootip will show the correct location.

Deleting the widget from the dashboard and then adding it again does not change the behaviour.

-------------
Issue #2:
The Visitor Location (City) widget's tootip does not show an apostrophe correctly.  Instead of showing the apostrophe, the tootip contains "&amp;#039;" .  The tootip in the Visitors in real-time widget, for the same location, shows the apostrophe correctly.  The location in question was "L'orignal, Ontario, Canada"
![issue 1 and 2](https://cloud.githubusercontent.com/assets/7869534/5332402/52323046-7e23-11e4-9132-71e125870318.jpg)

--------
Issue #3:
The Visitor Location (City) widget's tooltip does not always use the same presentation format for the location it displays.  Usually, the tootip text is presented in black inside a rectangle with rounded corners and a white background.  Occasionally, the tootip shows as white text on a black background inside a rectangle with no visible border but with rounded corners.  When this happens, the font size appears to be about 2 points larger than normal too.
![issue 3](https://cloud.githubusercontent.com/assets/7869534/5332404/5d1e5020-7e23-11e4-89f6-af09beb51ac6.jpg)

**Comment by canajun2eh** [comment]
Issue #1 does not appear to be related the location containing an apostrophe. It's happening fairly frequently on my site. 

The following strange behaviour has been observed:
When the Visitor Location (City) widget is displaying those three dots, refreshing the widget display by clicking on the "refresh" icon that appears when the cursor is in the title bar of the widget, will cause the locations to be properly displayed.  After that, when the mouse cursor is moved into the "business" part of the widget display, where the visitor locations are shown, the locations are replaced by those three dots.

**Comment by mattab** [comment]
maybe this is a bug in the Javascript truncation logic. can you somehow reproduce this issue on the http://demo.piwik.org ?

**Comment by canajun2eh** [comment]
Yes, all three issues are visible on the Piwik demo site too.

- Three dots immediately to the right of the country flag (and no "city" text)
- Tooltip sometimes white text on black background, and sometimes black text on white background
- apostrophe in city name not displayed as such in tooltip text.

**Comment by mattab** [comment]
Thanks @canajun2eh adding to next milestone. do you maybe have a link to demo showing the problem?

**Comment by canajun2eh** [comment]
Not necessary.

Simply add the Visitor Location (City) widget to the dashboard, and then view successive "pages" of this widget, 10 entries per "page".  Eventually, you'll see the 'three dots" problem, but you should try moving the mouse pointer into the  "cities" list, since that action seems to trigger the problem.  The problem first appears in the 41-50 page.

The varying tooltip presentation (white on black vs. black on white) will happen almost right away.  There doesn't seem to be any rhyme or reason to this problem.  This problem seems to go away altogether when I have used the "refresh" button in the title bar of the widget.  Before having used the "refresh" button, I could call up the tootip both when hovering over the country flag and also when hovering over the city name.  After the "refresh", the tootip only appears when hovering on the country  flag.

The incorrect presentation of an apostrophe in the tooltip will happen on any city name that contains an apostrophe.  I have noticed that the tooltip is OK when the mouse pointer hovers over the city name, but is incorrect when the pointer hovers over the country flag.

**Comment by tsteur** [comment]
Which browser are you using? I can't reproduce the 3 dots problem. Are you using the latest Piwik 2.11 version?

**Comment by canajun2eh** [comment]
I'm using Firefox 35.0.1 on a Windows 7 laptop.  The problem was observed on the demo Piwik site when Piwik 2.10 was current.  On the demo site, I had set the Visitor Location widget to display 10 locations per "page", and the "three dots" problem showed up on "page" 4 (locations 41 - 50).  On the demo site, you actually have to move the mouse cursor into the location list for the problem to show up.

Today, it happens on the demo site on "page" 7 (entries 71-80), when using Firefox but not when using Internet Explorer, Opera, Safari, or Google Chrome.

When I first submitted the problem, the "problem location" contained an apostrophe.  I don't think that this is a deciding factor.
